src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -407,6 +384,46 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
         return new $class();
     }
 
+    /**
+     * @param mixed                $parameterData
+     * @param string               $key
+     * @param \ReflectionParameter $constructorParameter
+     * @param array                $context
+     * @param string|null          $format
+     *
+     * @return mixed value of constructor parameter - an argument
+     *
+     * @throws LogicException
+     * @throws RuntimeException
+     * @throws MissingConstructorArgumentsException
+     */
+    protected function createConstructorArgument($parameterData, string $key, \ReflectionParameter $constructorParameter, array &$context, string $format = null)
+    {
+        if (null === $parameterData && $constructorParameter->allowsNull()) {
+            return null;
+        }
+        try {
+            if (null !== $constructorParameter->getClass()) {
```

Comment:
looks like this could be turned to 
```
if (null === $constructorParameter->getClass()) {
    return $parameterData;
}
```
and allow unindenting the block currently inside the "if"?